features/invitelist/impl/src/main/kotlin/io/element/android/features/invitelist/impl/InviteListState.kt

```diff
@@ -18,9 +18,20 @@ package io.element.android.features.invitelist.impl
 
 import androidx.compose.runtime.Immutable
 import io.element.android.features.invitelist.impl.model.InviteListInviteSummary
+import io.element.android.libraries.architecture.Async
+import io.element.android.libraries.matrix.api.core.RoomId
 import kotlinx.collections.immutable.ImmutableList
 
 @Immutable
 data class InviteListState(
-    val inviteList: ImmutableList<InviteListInviteSummary>
+    val inviteList: ImmutableList<InviteListInviteSummary>,
+    val confirmDialog: InviteConfirmationDialog = InviteConfirmationDialog.Hidden,
+    val acceptedAction: Async<RoomId> = Async.Uninitialized,
+    val declinedAction: Async<Unit> = Async.Uninitialized,
+    val eventSink: (InviteListEvents) -> Unit = {}
 )
+
+sealed interface InviteConfirmationDialog {
```

Comment:
For clarity maybe rename to 
```suggestion
sealed interface InviteDeclineConfirmationDialog {
```

`InviteListState.confirmDialog` could be renamed too.